pkg/alert/alert.go

```diff
@@ -0,0 +1,427 @@
+		}
+		if err := client.PostMessage(channel, fmt.Sprintf("%s```%s```", messagePrefix, messageSplitItem)); err != nil {
+			return err
+		}
+		time.Sleep(1200 * time.Millisecond) // need to wait 1 sec due to slack limits
+	}
+	alert.log.Info("Sent slack alert message of %d failing tests", len(failedTests))
+	alert.filePostedAlerts(failedTests)
+	return nil
+}
+
+//createMessage creates the alert message
+func createMessage(failedTests map[string]TestDetails, alert *Alert) string {
+	sortedKeys := make([]string, 0, len(failedTests))
+	for k := range failedTests {
+		sortedKeys = append(sortedKeys, k)
+	}
+	sort.Strings(sortedKeys)
+
+	writer := &strings.Builder{}
+	table := tablewriter.NewWriter(writer)
+	header := []string{"Test", "Landscape", "Provider", "K8s Ver", "OS", "Success", "Alert Reason", "Last failure"}
+
+	content := make([][]string, 0)
+	row := 0
+	for _, mapKey := range sortedKeys {
+		test := failedTests[mapKey]
+		failureReason := ""
+		if test.SuccessRate < float64(alert.cfg.SuccessRateThresholdPercent) {
+			failureReason = fmt.Sprintf("success rate < %d%%", alert.cfg.SuccessRateThresholdPercent)
+		} else if test.FailedContinuously {
+			failureReason = fmt.Sprintf(">%d failures in row", alert.cfg.ContinuousFailureThreshold-1)
+		}
+		newRow := []string{test.Name,
+			test.Landscape,
+			stringOrDefault(test.Cloudprovider, "-"),
+			stringOrDefault(test.K8sVersion, "-"),
+			stringOrDefault(test.OperatingSystem, "-"),
+			fmt.Sprintf("%d%%", int(test.SuccessRate)),
+			failureReason,
+			test.LastFailedTimestamp,
+		}
+		content = append(content, newRow)
+		row++
+	}
+
+	table.SetHeader(header)
+	table.SetAlignment(tablewriter.ALIGN_CENTER)
+	table.AppendBulk(content)
+	table.Render()
+	return writer.String()
+}
+
+//splitSlackMessage split message line wise based on given characters limit
+func splitSlackMessage(message string, charactersLimit int) []string {
+	var messageSplits []string
+	lines := strings.Split(message, "\n")
+	messageSplitItem := ""
+	for _, line := range lines {
+		if len(messageSplitItem)+len(line) < charactersLimit {
+			messageSplitItem += "\n" + line
+		} else {
+			messageSplits = append(messageSplits, messageSplitItem)
+			messageSplitItem = line
+		}
+	}
+	messageSplits = append(messageSplits, messageSplitItem)
+	return messageSplits
+}
+
+func stringOrDefault(input, defaultValue string) string {
+	if input == "" {
+		return defaultValue
+	}
+	return input
+}
+
+//elasticRequest send HTTP request to elasticsearch
+func (alert *Alert) elasticRequest(urlAttributes, httpMethod, payloadFormated string, result interface{}) {
+	url := fmt.Sprintf("%s%s", alert.cfg.Elasticsearch.Endpoint, urlAttributes)
+	payload := strings.NewReader(payloadFormated)
+	alert.log.Info(fmt.Sprintf("creating HTTP %s request on %s", httpMethod, url))
+	req, err := http.NewRequest(strings.ToUpper(httpMethod), url, payload)
+	if err != nil {
+		alert.log.Error(err, fmt.Sprintf("failed to create a http request for %s", url))
```

Comment:
return error. we do not want to continue with request if we are unable to create the request.